src/DurableTask.Core/TaskOrchestrationDispatcher.cs

```diff
@@ -425,12 +425,22 @@ protected async Task<bool> OnProcessWorkItemAsync(TaskOrchestrationWorkItem work
                                     break;
                                 case OrchestratorActionType.CreateSubOrchestration:
                                     var createSubOrchestrationAction = (CreateSubOrchestrationAction)decision;
+                                    DistributedTraceContext? traceContext = null;
+                                    if (traceActivity != null)
+                                    {
+                                        traceContext = new DistributedTraceContext(
+                                            traceActivity.TraceId.ToString(),
```

Comment:
`traceparent` is the id, not the traceid.

```suggestion
                                            traceActivity.Id.ToString(),
```